### PR TITLE
Add more JSON utilities

### PR DESCRIPTION
**Pull request by satabin**
In order to make the full roundtrip, this adds:
 - pretty/compact printing of Json streams
 - tokenizing ASTs

### CODE REVIEW

File: json/src/fs2/data/json/internal/Renderer.scala [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  }
+
+  def result: String = builder.result()
+
+}
+
+private[json] object Renderer {
+
+  def pipe[F[_]](pretty: Boolean, indent: String): Pipe[F, Token, String] = {
+    val builder = new Renderer(pretty, true, indent)
```
Comment (ybasket) [inline json/src/fs2/data/json/internal/Renderer.scala:167]: I have the suspicion this isn't safe 🤔

If you acquire this pipe once and then use it in concurrently for several streams, I feel like these could interfere in the local state of the `Renderer`. To be sure, the renderer should probably be acquired effectfully or at least suspended:

```scala
in => Stream.suspend(Stream.pure(new Renderer(pretty, true, indent))).flatMap { renderer =>
  in.mapChunks { chunk =>
       renderer += chunk
       Chunk.singleton(renderer.result)
     }
}
```

Comment (satabin) [reply json/src/fs2/data/json/internal/Renderer.scala:167]: You are correct, mister, I'll fix this.

Comment (satabin) [reply json/src/fs2/data/json/internal/Renderer.scala:167]: I wonder if I can't just
```scala
    { in =>
      val renderer = new Renderer(pretty, true, indent)
      in.mapChunks { chunk =>
       renderer += chunk
       Chunk.singleton(renderer.result)
      }
    }
```

Comment (satabin) [reply json/src/fs2/data/json/internal/Renderer.scala:167]: Nevermind, of course I can’t…

File: json/src/fs2/data/json/internal/Renderer.scala [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+  private def indentation(newline: Boolean, level: Int): Unit =
+    if (newline) {
+      builder.append('\n')
+      builder.append(indent * level)
+    }
+
+  private def jsonString(s: String, key: Boolean): Unit = {
+    @tailrec
+    def loop(idx: Int): Unit =
```
Comment (satabin) [inline json/src/fs2/data/json/internal/Renderer.scala:49]: @ybasket I also fixed escaping, but for the sake of "you never know what you do on a Saturday morning, so better safe than sorry", can you quickly check this new version?

File: json/src/fs2/data/json/package.scala [hunk 3]
```diff
@@ -58,6 +58,66 @@ package object json {
...
+
+    /** Renders a pretty-printed representation of the token stream with the given
+      * indentation size.
+      *
+      * Chunks can be concatenated to render all values in the stream,
+      * separated by new lines.
+      *
+      * You can use this to write the Json stream to a file.
+      */
+    def pretty[F[_]](indent: Int = 2): Pipe[F, Token, String] =
```
Comment (ybasket) [inline json/src/fs2/data/json/package.scala:89]: Both for here and the collector: Maybe let's not choose a side on the spaces vs tabs war and allow to specify the indentation string as another parameter with default? Maybe also the newline? No need to reproduce the full power of circe's pretty printer or similar, but these options seem very straight-forward to add…

File: json/src/fs2/data/json/internal/Renderer.scala (hunk 2 above)
Comment (ybasket) [reply json/src/fs2/data/json/internal/Renderer.scala:49]: It's also Saturday morning for me ;)

But looks all reasonable!

File: json/src/fs2/data/json/package.scala (hunk 3 above)
Comment (satabin) [reply json/src/fs2/data/json/package.scala:89]: :sob: Ok…

**Pull request merged**